clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,458 @@
+use core::str;
+use std::{
+    fs::{self, OpenOptions},
+    io::{self, Write},
+    process::{self, Command, Stdio},
+};
+
+use const_format::concatcp;
+use rust_cgi::{http::Method, http::StatusCode, text_response, Request, Response};
+use tracing::warn;
+
+use crate::{
+    command_listener::CommandResponse,
+    config::{TeliodDaemonConfig, TeliodDaemonConfigPartial},
+    ClientCmd, DaemonSocket, TeliodError, TIMEOUT_SEC,
+};
+
+const TELIOD_TMP_DIR: &str = "/tmp/nordsecuritymeshnet";
+const PID_FILE: &str = concatcp!(TELIOD_TMP_DIR, "/teliod.pid");
+const QPKG_DIR: &str = "/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet";
+const TELIOD_BIN: &str = concatcp!(QPKG_DIR, "/teliod");
+const MESHNET_LOG: &str = concatcp!(QPKG_DIR, "/meshnet.log");
+const TELIOD_LOG: &str = "/var/log/teliod.log";
+
+#[cfg(not(test))]
+const TELIOD_CFG: &str = concatcp!(QPKG_DIR, "/teliod.cfg");
+#[cfg(test)]
+use tests::TELIOD_CFG;
+
+pub struct PidFile;
+
+impl PidFile {
+    pub async fn new() -> Result<Self, TeliodError> {
+        let mut file = OpenOptions::new()
+            .write(true)
+            .create(true)
+            .truncate(true)
+            .open(PID_FILE)?;
+        writeln!(file, "{}", process::id())?;
+        Ok(Self)
+    }
+}
+
+impl Drop for PidFile {
+    fn drop(&mut self) {
+        if let Err(e) = std::fs::remove_file(PID_FILE) {
+            warn!("Failed to remove PID file: {}", e);
+        }
+    }
+}
+
+macro_rules! teliod_blocking_query {
+    ($command:expr) => {{
+        let socket_path = DaemonSocket::get_ipc_socket_path();
+        futures::executor::block_on(tokio::time::timeout(
+            std::time::Duration::from_secs(TIMEOUT_SEC),
+            DaemonSocket::send_command(
+                &socket_path.unwrap_or_default(),
+                &serde_json::to_string(&$command).unwrap_or_default(),
+            ),
+        ))
+    }};
+}
+
+pub(crate) fn handle_request(request: Request) -> Response {
+    if request.uri().path() != "/cgi-bin/qpkg/teliod.cgi" {
```

Comment:
Is this really correct ?

Path should be string after "/cgi-bin/qpkg/teliod.cgi'

Reply:
path verification is removed